R/fit_peak.R

```diff
@@ -0,0 +1,173 @@
+#' Fit a peak algorithm to weighted time series observations.
+#'
+#' @description
+#' `r lifecycle::badge("stable")`
+#'
+#' This function calculates the intensity levels of peak time series
+#' observations to the next comming season based on previouse seasons.
+#' Peak observations are the n highest observations from each season.
+#' It provides low, medium, high intensity levels, that reflect the
+#' current intensity of infection based on previous seasons. Hence it
+#' uses the peak observations from each previous season it can only
+#' explain the current intensity in relation to the peak of the season.
+#'
+#' @param weighted_observations A tibble containing two columns of size n;
+#' `observation`, which represents the data points, and `weight`, which is the
+#' importance assigned to an observation. Higher weights indicate that an
+#' observation has more influence on the model outcome, while lower weights
+#' reduce its impact. The structure is:
+#' `weighted_observations = tibble(observation = c(...), weight = c(...))`
```

Comment:
You already write in words that it should have this structure. Due to how the param fields are rendered in the documentation, I think you would be better off not repeating the structure like this -- even though it makes it very explicit.